### PR TITLE
Extract L1B DE function for better code modularization

### LINKED ISSUE #968: Hi L1B - extract method for generating l1b_de DataArrays
### Algorithm Description:

Extracting this method will help modularize code and make testing of individual l1b de functions more straightforward.

### PR DESCRIPTION
**Pull request by tmplummer**
# Change Summary

## Overview
While working on another ticket, I decided that this would help make testing easier so I implemented that change in it's own ticket to keep the PR small.
- Extract l1b function that instantiates the xarray.DataArrays for the L1B de product
- Add test coverage for newly extracted function

Closes: #968

### CODE REVIEW

File: poetry.lock
Comment (tmplummer) [inline poetry.lock:1]: I'm still not quite sure about this change. Last time I muddled with dependencies, Greg indicated I had done something wrong. This time I let the pre-commit hook modify the lock file before committing.

File: pyproject.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ classifiers = [
 cdflib = "==1.2.6"
 imap-data-access = ">=0.5.0"
 python = ">=3.9,<4"
-space_packet_parser = ">=4.2.0"
+space_packet_parser = "^4.2.0"
```
Comment (tmplummer) [inline pyproject.toml]: The release of `space_packet_parser 5.0.0` has breaking changes so for now I pinned the version less than 5.

File: imap_processing/hi/utils.py [hunk 2]
```diff
@@ -25,3 +32,94 @@ def sensor(self) -> str:
...
+        Coordinate variables for the Dataset.
+    shape : int or tuple
+        Shape of ndarray data array to instantiate in the xarray.DataArray.
+
+    Returns
+    -------
+    data_array : xarray.DataArray
+        Meeting input specifications.
+    """
+    _attrs = attrs.copy()
```
Comment (laspsandoval) [inline imap_processing/hi/utils.py:66]: I think that you meant to use _attrs in the rest of the function after this line.

File: imap_processing/tests/hi/test_utils.py [hunk 3]
```diff
@@ -13,3 +18,57 @@ def test_hiapid():
...
+
+@pytest.mark.parametrize(
+    "var_names, shape, lookup_str",
+    [
+        (["delta_t_ab", "delta_t_ac1"], 5, "hi_de_{0}"),
+        (["hae_latitude"], (3, 5), "hi_pset_{0}"),
+    ],
+)
+def test_create_dataset_variables(var_names, shape, lookup_str):
+    """Test coverage for `imap_processing.hi.utils.create_l1b_de_variables`"""
```
Comment (laspsandoval) [inline imap_processing/tests/hi/test_utils.py]: Docstring name change.

File: imap_processing/tests/hi/test_utils.py [hunk 4]
```diff
@@ -13,3 +18,57 @@ def test_hiapid():
...
+
+
+@pytest.mark.parametrize(
+    "var_names, shape, lookup_str",
+    [
+        (["delta_t_ab", "delta_t_ac1"], 5, "hi_de_{0}"),
+        (["hae_latitude"], (3, 5), "hi_pset_{0}"),
+    ],
+)
+def test_create_dataset_variables(var_names, shape, lookup_str):
```
Comment (laspsandoval) [inline imap_processing/tests/hi/test_utils.py:54]: You passed in shape and then assigned a value to shape=5.

**Review by laspsandoval — APPROVED** [review]
Nice utils. A few changes and then I think you can merge.

File: poetry.lock
Comment (laspsandoval) [reply poetry.lock:1]: @maxinelasp set up the dependencies. Maybe ping her about it if you have questions.

File: imap_processing/hi/utils.py (hunk 2 above)
Comment (tmplummer) [reply imap_processing/hi/utils.py:66]: Good catch!

**Pull request merged**